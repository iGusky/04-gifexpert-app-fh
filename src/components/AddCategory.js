import React from 'react'
import { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ( {setCategorias} ) => {

  const [inputValue, setInputValue ] = useState('');

  const handleInputChange = (e) => {
    setInputValue(e.target.value);
  }
 
  const handleSubmit = (e) =>{
    e.preventDefault();
    if(inputValue.trim().length>2){
      setCategorias( cats => [ inputValue, ...cats]);
      setInputValue('');
    }
  }

  return (
    <form onSubmit={ handleSubmit } >
        <p>{inputValue}</p>
      <input
        type = "text"
        value = {inputValue}
        onChange = {(e) => handleInputChange(e)}
      />
    </form>
  )
}

AddCategory.propTypes = {
  setCategorias: PropTypes.func.isRequired
};
