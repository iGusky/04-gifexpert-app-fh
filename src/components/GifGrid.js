import React from 'react'
// import {useState, useEffect} from 'react';
// import { getGifs } from '../helpers/getGifs';
import { GifGridItem } from './GifGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGrid = ({ category }) => {

    //*Antes de usar Customhooks
    // const [images, setImages] = useState([]);
    //Con este codigo hacemos que el getGifs se ejecute solo cuando se rendereiza por primera vez.
    // useEffect( () => {
    //     getGifs( category )
    //         .then( imgs => setImages(imgs));
    // }, [ category ]); //Si la categoria cambia, vuelve a ejecutar el efecto interno.

    //* Usando Customhooks
    //Se renombra data a images
    const {data:images, loading } = useFetchGifs( category );

    return (
    <>
        <h3 className="animate__animated animate__fadeIn">{ category }</h3>

        {/* Forma corta de un if */}
        { loading && <p className="animate__animated animate__flash">Cargando...</p>}

        <div className="card-grid">
            {
                images.map((img) => {
                    return <GifGridItem
                        key = {img.id}
                        {...img}
                    />
                })
            }
        </div>
    </>
    )
}