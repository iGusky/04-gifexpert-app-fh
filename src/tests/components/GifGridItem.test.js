import { shallow } from 'enzyme';
import '@testing-library/jest-dom';

import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas de <GifGridItem />', () => {

    const titulo = 'Un titulo cualquiera';
    const url = 'https://localhost/imagen.jpg';
    let wrapper = shallow(<GifGridItem title = { titulo } url = { url }/>);

    it('Debe renderizar el componente <GifGridItem /> correctamente',() => {
        expect(wrapper).toMatchSnapshot();
    });

    it('Debe tener el pÃ¡rrafo con el titulo', () => {
        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe( titulo );
    });

    it('Debe tener la imagen igual al url y alt  de los props', () => {
        const img = wrapper.find('img');
        expect( img.prop('src') ).toBe( url );
        expect( img.prop('alt') ).toBe( titulo );
    });

    it('El div, debe tener la clase "animate__fadeIn"',() => {
        const div = wrapper.find('div');
        expect( div.hasClass('animate__fadeIn') ).toBe(true);
    })
 
});
