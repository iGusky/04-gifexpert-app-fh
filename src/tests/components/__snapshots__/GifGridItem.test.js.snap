// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <GifGridItem /> Debe renderizar el componente <GifGridItem /> correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un titulo cualquiera"
    src="https://localhost/imagen.jpg"
  />
  <p>
    Un titulo cualquiera
  </p>
</div>
`;
